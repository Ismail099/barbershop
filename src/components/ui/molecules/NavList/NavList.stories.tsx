import { Story, Meta } from '@storybook/react';

import { NavList } from './NavList';

export default {
  title: 'molecules/NavList',
  component: NavList,
  layout: 'fullscreen',
  parameters: {
    backgrounds: {
      default: 'dark',
      values: [{ name: 'dark', value: '#121212' }]
    }
  }
} as Meta;

export const Basic: Story = () => <NavList />;

export const Mobile: Story = () => <NavList />;

Mobile.parameters = {
  layout: 'fullscreen',
  viewport: {
    defaultViewport: 'mobile1'
  }
};
