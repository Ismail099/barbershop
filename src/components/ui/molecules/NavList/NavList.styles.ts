import styled from 'styled-components';
import media from 'styled-media-query';

export const NavList = styled.ul`
  position: relative;
  width: 100%;
  height: 100%;
  list-style: none;
  display: flex;
  align-items: center;
  justify-content: space-between;
  overflow-x: hidden;

  ${media.greaterThan('small')`
    justify-content: center;
  `}
`;
