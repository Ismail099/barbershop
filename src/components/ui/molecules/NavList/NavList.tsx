import { useState } from 'react';

import Fade from 'react-reveal/Fade';
import { Sling as Hamburger } from 'hamburger-react';

import MediaMatch from '../MediaMatch/MediaMatch';
import { Logotipo } from '~atoms/Logotipo/Logotipo';

import * as S from './NavList.styles';
import { MenuMobile } from '../MenuMobile/MenuMobile';

export const NavList = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <S.NavList>
      <Fade left>
        <Logotipo hideOnMobile />
      </Fade>

      <MediaMatch lessThan="medium">
        <Fade right>
          <Hamburger
            color="#FFF"
            duration={0.4}
            size={30}
            distance="md"
            rounded
            direction="right"
            easing="ease-out"
            toggled={isOpen}
            toggle={setIsOpen}
            label="Abrir e Fechar"
          />

          <MenuMobile isOpen={isOpen} />
        </Fade>
      </MediaMatch>
    </S.NavList>
  );
};
