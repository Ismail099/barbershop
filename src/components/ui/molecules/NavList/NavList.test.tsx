import { renderWithTheme } from '~src/utils/tests/helpers';

import { NavList } from './NavList';

describe('<NavList />', () => {
  it('Deve renderizar o componente <NavList />', () => {
    //Arrange
    const { container } = renderWithTheme(<NavList />);
    //Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
