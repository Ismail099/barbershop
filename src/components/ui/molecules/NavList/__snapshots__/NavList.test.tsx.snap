// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavList /> Deve Renderizar todos os components de <NavList /> 1`] = `
.c1 {
  display: none;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100%;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow-x: hidden;
}

.c2 {
  background: #080808;
  opacity: 0;
  pointer-events: none;
  -webkit-transition: 0.3s linear;
  transition: 0.3s linear;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: calc(4.8rem + (1.5rem + 1.5rem));
  z-index: 20;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow-y: auto;
  overflow-x: hidden;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c4 {
  position: relative;
  list-style: none;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4:first-child::after {
  width: 100%;
}

.c4:first-child {
  color: #F71812;
  text-shadow: 0px 0px 30px rgba(247,24,18,0.5),0px 0px 15px rgba(247,24,18,0.4),0px 0px 10px rgba(247,24,18,0.4),0px 0px 5px rgba(247,24,18,0.2);
}

.c4::after {
  content: '';
  position: absolute;
  justify-self: center;
  bottom: -0.5rem;
  width: 0;
  height: 0.2rem;
  background: #F71812;
  box-shadow: 0px 0px 10px #F71812;
  -webkit-transition: 0.3s linear;
  transition: 0.3s linear;
}

.c4:hover::after,
.c4:focus::after {
  width: 100%;
}

.c5 {
  font-size: 1.8rem;
  font-weight: 300;
  padding: 0 1rem;
  -webkit-transition: 0.3s linear;
  transition: 0.3s linear;
}

.c5:hover,
.c5:focus {
  color: #F71812;
  cursor: pointer;
  text-shadow: 0px 0px 30px rgba(247,24,18,0.5),0px 0px 15px rgba(247,24,18,0.4),0px 0px 10px rgba(247,24,18,0.4),0px 0px 5px rgba(247,24,18,0.2);
}

@media (max-width:768px) {
  .c1 {
    display: block;
  }
}

@media (min-width:450px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<ul
  class="c0"
>
  <div
    class="react-reveal"
  >
    <div
      data-testid="Logo"
    />
  </div>
  <div
    class="c1"
  >
    <div
      class="react-reveal"
    >
      <div
        aria-label="Abrir e Fechar"
        class="hamburger-react"
        role="button"
        style="cursor: pointer; height: 48px; position: relative; transition: 0.4s ease-out; user-select: none; width: 48px; outline: none; transform: none;"
        tabindex="0"
      >
        <div
          style="background: rgb(255, 255, 255); height: 3px; left: 9px; position: absolute; border-radius: 9em; width: 30px; top: 13px; transition: 0.4s ease-out; transform: none;"
        />
        <div
          style="background: rgb(255, 255, 255); height: 3px; left: 9px; position: absolute; border-radius: 9em; width: 30px; top: 23px; transition: 0.4s ease-out; transform: none;"
        />
        <div
          style="background: rgb(255, 255, 255); height: 3px; left: 9px; position: absolute; border-radius: 9em; width: 30px; top: 33px; transition: 0.4s ease-out; transform: none;"
        />
      </div>
    </div>
    <div
      class="react-reveal"
    >
      <div
        aria-hidden="true"
        aria-label="Menu Full"
        class="c2"
      >
        <ul
          class="c3"
        >
          <li
            class="c4"
          >
            <a
              aria-label="Home"
              class="c5"
            >
              Home
            </a>
          </li>
          <li
            class="c4"
          >
            <a
              aria-label="Sobre"
              class="c5"
            >
              Sobre
            </a>
          </li>
          <li
            class="c4"
          >
            <a
              aria-label="Serviços"
              class="c5"
            >
              Serviços
            </a>
          </li>
          <li
            class="c4"
          >
            <a
              aria-label="Equipe"
              class="c5"
            >
              Equipe
            </a>
          </li>
          <li
            class="c4"
          >
            <a
              aria-label="Opinião"
              class="c5"
            >
              Opinião
            </a>
          </li>
          <li
            class="c4"
          >
            <a
              aria-label="FeedBack"
              class="c5"
            >
              FeedBack
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</ul>
`;
